Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1231,6 +1231,17 @@ DataDetectionResultsStorage& LocalFrame::dataDetectionResults()
 
 #endif
 
+void LocalFrame::frameWasDisconnectedFromOwner() const
+{
+    if (!m_doc)
+        return;
+
+    if (RefPtr window = m_doc->domWindow())
+        window->willDetachDocumentFromFrame();
+
+    m_doc->detachFromFrame();
```

Comment:
Can we protected the document before calling `detachFromFrame()` on it. This doesn't match our modern smart pointer idioms.

Reply:
`protectedDocument()->detachFromFrame();` (unless it is possible for the document to be in the middle of destruction).